metpy/calc/thermo.py

```diff
@@ -406,12 +406,14 @@ def lfc(pressure, temperature, dewpt, parcel_temperature_profile=None, dewpt_sta
     # The parcel profile and data may have the same first data point.
     # If that is the case, ignore that point to get the real first
     # intersection for the LFC calculation.
+    pressure_units = pressure.units
     if np.isclose(parcel_temperature_profile[0].m, temperature[0].m):
-        x, y = find_intersections(pressure[1:], parcel_temperature_profile[1:],
+        x, y = find_intersections(np.log(pressure[1:].m), parcel_temperature_profile[1:],
```

Comment:
Wondering if this functionality should be an option on `find_intersections`. Thoughts?

Reply:
We could put it there. I assume that pressure would be the only x variable that would need this treatment, so it should be simple enough to add that. I would guess there's other uses for `find_intersections` in the community, so this may be a worthwhile option to have.

Reply:
Pressure would be the only one we need it for, but I'd put it as an option, maybe "log_x", that tells it to log transform the x data. I'm 50-50 on it, but it would reduce some duplicated code.